python/pyiceberg/cli/output.py

```diff
@@ -0,0 +1,155 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+from abc import ABC, abstractmethod
+from typing import Any, List
+from uuid import UUID
+
+from rich.console import Console
+from rich.table import Table as RichTable
+from rich.tree import Tree
+
+from pyiceberg.schema import Schema
+from pyiceberg.table import Table
+from pyiceberg.table.partitioning import PartitionSpec
+from pyiceberg.typedef import Identifier, Properties
+
+
+class Output(ABC):
+    """Output interface for exporting"""
+
+    @abstractmethod
+    def exception(self, ex: Exception):
+        ...
+
+    @abstractmethod
+    def identifiers(self, identifiers: List[Identifier]):
+        ...
+
+    @abstractmethod
+    def describe_table(self, table):
+        ...
+
+    @abstractmethod
+    def describe_properties(self, properties: Properties):
+        ...
+
+    @abstractmethod
+    def text(self, response: str):
+        ...
+
+    @abstractmethod
+    def schema(self, schema: Schema):
+        ...
+
+    @abstractmethod
+    def spec(self, spec: PartitionSpec):
+        ...
+
+    @abstractmethod
+    def uuid(self, uuid: UUID):
+        ...
+
+
+class ConsoleOutput(Output):
+    def exception(self, ex: Exception):
+        Console(stderr=True).print(ex)
+
+    def identifiers(self, identifiers: List[Identifier]):
+        table = RichTable.grid()
+        for identifier in identifiers:
+            table.add_row(".".join(identifier))
+
+        Console().print(table)
+
+    def describe_table(self, table: Table):
+        assert table.metadata
+        metadata = table.metadata
+        table_properties = RichTable.grid()
+
+        for key, value in metadata.properties.items():
+            table_properties.add_row(key, value)
+
+        schema_tree = Tree("Schema")
+        for field in metadata.current_schema().fields:
+            schema_tree.add(str(field))
+
+        snapshot_tree = Tree("Snapshots")
+        for snapshot in metadata.snapshots:
+            snapshot_tree.add(f"Snapshot {snapshot.schema_id}: {snapshot.manifest_list}")
+
+        output_table = RichTable.grid()
+        output_table.add_row("Table format version", str(metadata.format_version))
+        output_table.add_row("Metadata location", table.metadata_location)
+        output_table.add_row("Table UUID", str(table.metadata.table_uuid))
+        output_table.add_row("Last Updated", str(metadata.last_updated_ms))
+        output_table.add_row("Partition spec", str(metadata.current_partition_spec()))
+        output_table.add_row("Sort order", str(metadata.current_sort_order()))
+        output_table.add_row("Schema", schema_tree)
+        output_table.add_row("Snapshots", snapshot_tree)
+        output_table.add_row("Properties", table_properties)
+        Console().print(output_table)
+
+    def describe_properties(self, properties: Properties):
+        output_table = RichTable.grid()
```

Comment:
I think you need to add `padding=2` to this, or else you get columns that run together:

```
Welcome to PyIceberg 🐧
write.delete.parquet.compression-codecgzip
write.metadata.compression-codec      gzip
...
```

Reply:
`padding=2` gives:
```
➜  python git:(fd-python-cli) ✗ pyiceberg properties fokko.table   
Welcome to PyIceberg 🐧
write.delete.parquet.compression-codec  zstd
                                            
                                            
write.metadata.compression-codec        gzip
                                            
                                            
write.summary.partition-limit           100 
                                            
                                            
write.parquet.compression-codec         zstd
```

I think we want to go for `padding=(0, 2)`:
```
➜  python git:(fd-python-cli) ✗ pyiceberg properties fokko.table
Welcome to PyIceberg 🐧
write.delete.parquet.compression-codec  zstd
write.metadata.compression-codec        gzip
write.summary.partition-limit           100 
write.parquet.compression-codec         zstd
```